test/e2e_test.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+set -e
+
+SERVER_PORT=9991
+FILTER_PORT=9992
+
+pushd ".."
+cargo build
+popd
+
+function tag {
+  local TAG=$1
+  while read LINE
+  do
+    echo "$TAG" "$LINE"
+  done
+}
+
+python3 -u test-server.py 127.0.0.1 "$SERVER_PORT" 2>&1 | tag "[FILTER]" &
+
+RUST_BACKTRACE=1 \
+RUST_LOG='jsonrpc_filter' \
+../target/debug/jsonrpc-filter \
+  --allowed-list ./test_allowed.txt \
+  --bind 127.0.0.1 --port $FILTER_PORT \
+  --forward "http://127.0.0.1:$SERVER_PORT" 2>&1 | tag "[SERVER]" &
+
+function finish {
+  kill $(jobs -p) || true
```

Comment:
Why is ` || true` needed?

Reply:
processes may die between `jobs` and `kill`. `kill` will complain that 'there is no such process', and because of `set -e`, the script halts its execution there. Should I `set +e` on the exit trap?

Reply:
I'm confusing about the behavior of `set -e`. Doesn't it make `e2e_test.sh` stop when `jobs` exits with non-zero?

Reply:
Yes it does. `set -e` turns on the behavior, and `set +e` turns off it.

Reply:
Well, if it does, why do you need `|| true`? The script will terminate when jobs exit with a non-zero value.

Reply:
I wanted to make sure `wait` is called. It waits for child processes to clean up and reports whether they are terminated properly.